arduino/WandCore/WandCore.ino

```diff
@@ -143,11 +152,17 @@ void setup()
       debugI("Start updating %d", type);
     })
     .onEnd([]() {
-      debugI("/nEnd");
+      debugI("OTA upload complete. Rebooting...");
     })
     .onProgress([](unsigned int progress, unsigned int total) {
-      // KNOWN ISSUE - OTAing while telnet is open will cause the upload to fail. Need to implement throttling.
-      debugI("Progress: %d, Total: %d", progress, total);
+      int bigprog = 100 * progress;
+      int percent = bigprog / total;
+      int permod = percent % 10;
+      if (percent % 10 == 0 && percent > otaProgress) 
+      {
+        otaProgress = percent;
+        debugI("OTA upload is %d percent complete", percent);
+      }

```

Comment:
I'm not super familiar with the logic here, can you add some comments? Also, I feel like `bigprog` is not a great variable name.

Reply:
Does this get called for every int `progress`? (`1, 2, 3...`) I imagine this is in bytes.
If it doesn't then the `percent % 10 == 0` wouldn't always work, so I assume it does.

Reply:
if you mod two ints and they don't divide evenly, it truncates the decimals off. added some comments and changed the variable name.

Reply:
It's not about the modulo operation, it's about the input `percent `. This conditional logic is only called if `percent = 10`, `percent = 20`, `percent = 30`...
If this functional is called after a random number of bytes are uploaded (e.g. `percent = 8`, `percent = 11`, `percent = 12`...), you may miss the case where `percent % 10 == 0`.

Reply:
I see what you are saying in theory. In practice, this top level call (ArduinoOTA.onProgress) is really chatty, and what ends up happening is that there are like a dozen cases where `percent = 10` (or `20`, `30`, etc), hence the other logic of only actually logging it once. 

Worst case, it skips over one of the reports, which would be relatively fine imo.

Reply:
Yeah, that's what I figured. Cool cool